spawn_sdk/spawn_sdk/lib/defact.ex

```diff
@@ -19,6 +19,66 @@ defmodule SpawnSdk.Defact do
     end
   end
 
+  defmacro action(action_name, opts, block_fn) do
+    action_name = parse_action_name(action_name)
+
+    define_action(action_name, block_fn, opts)
+  end
+
+  defmacro action(action_name, block_fn) do
+    action_name = parse_action_name(action_name)
+
+    define_action(action_name, block_fn)
+  end
+
+  defp define_action(action_name, block_fn, opts \\ []) do
+    quote do
+      Module.put_attribute(
+        __MODULE__,
+        :defact_exports,
+        Macro.escape({unquote(action_name), %{timer: Keyword.get(unquote(opts), :timer)}})
+      )
+
+      def handle_action({unquote(action_name), payload}, context) do
+        try do
+          case :erlang.fun_info(unquote(block_fn), :arity) do
+            {:arity, 0} ->
```

Comment:
I don't know if it's worth providing a function with arity 0 because in fact the user can use the context in a way not entirely related to the object of the request